src/OctoshiftCLI.IntegrationTests/GhesToGithub.cs

```diff
@@ -36,45 +38,60 @@ public GhesToGithub(ITestOutputHelper output)
-        };
+        _sourceGithubToken = Environment.GetEnvironmentVariable("GHES_PAT");
+        _targetGithubToken = Environment.GetEnvironmentVariable("GHEC_PAT");
+        _azureStorageConnectionString = Environment.GetEnvironmentVariable($"AZURE_STORAGE_CONNECTION_STRING_GHES_{TestHelper.GetOsName().ToUpper()}");
 
         _versionClient = new HttpClient();
         _archiveUploader = new ArchiveUploader(_targetGithubClient, logger);
 
         _sourceGithubHttpClient = new HttpClient();
-        _sourceGithubClient = new GithubClient(logger, _sourceGithubHttpClient, new VersionChecker(_versionClient, logger), new RetryPolicy(logger), new DateTimeProvider(), sourceGithubToken);
+        _sourceGithubClient = new GithubClient(logger, _sourceGithubHttpClient, new VersionChecker(_versionClient, logger), new RetryPolicy(logger), new DateTimeProvider(), _sourceGithubToken);
         _sourceGithubApi = new GithubApi(_sourceGithubClient, GHES_API_URL, new RetryPolicy(logger), _archiveUploader);
 
         _targetGithubHttpClient = new HttpClient();
-        _targetGithubClient = new GithubClient(logger, _targetGithubHttpClient, new VersionChecker(_versionClient, logger), new RetryPolicy(logger), new DateTimeProvider(), targetGithubToken);
+        _targetGithubClient = new GithubClient(logger, _targetGithubHttpClient, new VersionChecker(_versionClient, logger), new RetryPolicy(logger), new DateTimeProvider(), _targetGithubToken);
         _targetGithubApi = new GithubApi(_targetGithubClient, "https://api.github.com", new RetryPolicy(logger), _archiveUploader);
 
-        _blobServiceClient = new BlobServiceClient(azureStorageConnectionString);
+        _blobServiceClient = new BlobServiceClient(_azureStorageConnectionString);
 
         _sourceHelper = new TestHelper(_output, _sourceGithubApi, _sourceGithubClient) { GithubApiBaseUrl = GHES_API_URL };
         _targetHelper = new TestHelper(_output, _targetGithubApi, _targetGithubClient, _blobServiceClient);
     }
 
-    public async Task Basic()
+    [Theory]
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task Basic(bool useGithubStorage)
     {
         var githubSourceOrg = $"e2e-testing-{TestHelper.GetOsName()}";
         var githubTargetOrg = $"octoshift-e2e-ghes-{TestHelper.GetOsName()}";
         const string repo1 = "repo-1";
         const string repo2 = "repo-2";
 
+        var tokens = new Dictionary<string, string> { };
+
+        tokens = useGithubStorage
+            ? new Dictionary<string, string>
+            {
+                ["GH_SOURCE_PAT"] = _sourceGithubToken,
+                ["GH_PAT"] = _targetGithubToken,
+            }
+            : new Dictionary<string, string>
+            {
+                ["GH_SOURCE_PAT"] = _sourceGithubToken,
+                ["GH_PAT"] = _targetGithubToken,
+                ["AZURE_STORAGE_CONNECTION_STRING"] = _azureStorageConnectionString
+            };
```

Comment:
no need to do this, you can leave everything as it was before (initializing in the constructor) and just remove the `AZURE_STORAGE_CONNECTION_STRING` from it and then in the test method conditionally add it:

```suggestion
        if (!useGithubStorage)
        {
            _tokens["AZURE_STORAGE_CONNECTION_STRING"] = _azureStorageConnectionString;
        }
```